compiler/src/dotty/tools/dotc/transform/GenericSignatures.scala

```diff
@@ -283,7 +283,7 @@ object GenericSignatures {
           jsig(atp, toplevel, primitiveOK)
 
         case hktl: HKTypeLambda =>
-          jsig(hktl.finalResultType)
+          jsig(hktl.finalResultType, toplevel, primitiveOK)
```

Comment:
recursive methods with default parameters are dangerous and I almost think we should warn when this happens. In any case, I suggest removing the default parameters from this method to avoid this issue in the future.

Reply:
Most calls seem to intentionally drop the params. Many times that's appropriate, some times I'm not sure.
Sadly too many other deadlines to review this in depth right now...

Reply:
OK, let's keep as-is to minimize the diff with the scalac implementation then.